teps/0130-pipeline-level-build-cache.md

```diff
@@ -0,0 +1,557 @@
+  - [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Out of Scope](#out-of-scope)
+  - [Notes and Caveats](#notes-and-caveats)
+    - [Other uses for this feature](#other-uses-for-this-feature)
+    - [Related features in other CI/CD systems](#related-features-in-other-cicd-systems)
+- [Design Details](#design-details)
+- [Design Evaluation](#design-evaluation)
+  - [Reusability](#reusability)
+  - [Simplicity](#simplicity)
+  - [Flexibility](#flexibility)
+  - [Conformance](#conformance)
+  - [User Experience](#user-experience)
+  - [Performance](#performance)
+  - [Security](#security)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+  - [Improve our documentation](#improve-our-documentation)
+  - [Catalog Tasks for docker daemon](#catalog-tasks-for-docker-daemon)
+  - [First-class docker support](#first-class-docker-support)
+- [Implementation Plan](#implementation-plan)
+  - [Test Plan](#test-plan)
+  - [Infrastructure Needed](#infrastructure-needed)
+  - [Upgrade and Migration Strategy](#upgrade-and-migration-strategy)
+  - [Implementation Pull Requests](#implementation-pull-requests)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+This TEP proposes support for more complex build Pipelines by allowing multiple Tasks in
+a PipelineRun to share the same build cache; for example, using the same docker daemon.
+
+## Motivation
+
+The docker-build catalog Task only allows building a single image and pushing it to a registry.
+The following use cases are not addressed:
+- Building multiple images concurrently with the same daemon to save resources and share cached base layers
+- Introducing additional steps, such as testing, in between a build and a push
+  - A user may also want to guard guard execution of the push based on the results of those tasks
+  - A user may also want to use the built image in these subsequent tasks without having to first push to and pull from a registry;
+  for example, if using `docker exec` or `docker-compose`
+
+Allowing multiple Tasks in a PipelineRun to use the same docker daemon allows PipelineRuns with
+docker builds to take advantage of Pipeline features (e.g. using matrix to build multiple images,
+executing a push conditionally on previous steps) while still having access to the same local
+cache of built images and avoid having to push to a registry in between.
+
+### Goals
+
+- More flexible build Pipelines
+- Build cache is only accessible to the single PipelineRun that uses it
+
+### Non-Goals
+
+- Sharing a daemon or build cache between PipelineRuns
+
+### Requirements
+
+- A docker image built during a Pipeline can be used in a subsequent Task without first pushing it to an image registry
+
+## Proposal
+
+Add support for Sidecars with the same lifespan as PipelineRuns, for example:
+
+```yaml
+kind: Pipeline
+spec:
+  params:
+  - name: image
+  - name: docker-host
+    value: $(sidecars.docker-daemon.host):2376
```

Comment:
I don't think that's correct. `Pipeline` params do not take `value` but `default`. That `value` would probably need to be used directly when using the `Task`, so line 129 and so on.

Reply:
thanks, fixed.